benches/bigint.rs

```diff
@@ -73,7 +73,16 @@ fn u512_mul(b: &mut Bencher) {
 	b.iter(|| {
 		let n = black_box(10000);
 		let one = black_box(U512::one());
-		(1..n).fold(one, |old, new| { old.overflowing_mul(U512::from(black_box(new))).0 })
+		(1..n).fold(one, |old, new| { old.overflowing_mul(U512::from(black_box(new | 1))).0 })
+	});
+}
+
+#[bench]
+fn u512_mul_small(b: &mut Bencher) {
+	b.iter(|| {
+		let n = black_box(10000);
+		let one = black_box(U512::one());
+		(1..153).fold(one, |old, new| { old.overflowing_mul(U512::from(10)).0 })
```

Comment:
Sure, but as long as it can't be trivially converted to a shift I don't mind. I don't know what you mean by degenerate in this case, it can't overflow (or rather, shouldn't be able to, but the tests will catch that if it becomes untrue).